cfme/tests/cloud/test_keypairs.py

```diff
@@ -92,11 +87,12 @@ def test_keypair_create_cancel(openstack_provider):
         * Select Cloud Provider.
         * Also delete it.
     """
-    keypair = KeyPair(name=fauxfactory.gen_alphanumeric(), provider=openstack_provider)
-    keypair.create(cancel=True)
-
-    with pytest.raises(KeyPairNotFound):
-        navigate_to(keypair, 'Details')
+    keypairs = KeyPairCollection()
+    keypair = keypairs.create(name="", provider=openstack_provider, cancel=True)
+    try:
+        assert not keypair.exists
+    except KeyPairNotFound:
+        pass
```

Comment:
**REQUIRED**
`KeyPairNotFound` will never caught, just leave `assert not keypair.exists`